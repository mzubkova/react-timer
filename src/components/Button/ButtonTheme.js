import React, { useContext } from "react";
import PropTypes from "prop-types";

import { ThemeContext } from "../Context/ThemeContext";

import "./ButtonTheme.scss";

const ButtonTheme = () => {
  const { theme, setTheme } = useContext(ThemeContext);

  const toggleThemeMode = () => {
    setTheme(theme === "dark" ? "light" : "dark");
  };

  return (
    <button className="switcher-btn" onClick={toggleThemeMode}>
      {theme === "dark" ? "Light" : "Dark"} theme
    </button>
  );
};

ButtonTheme.propTypes = {
  onButtonClick: PropTypes.func,
};

export default ButtonTheme;
