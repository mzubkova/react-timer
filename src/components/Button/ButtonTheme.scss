.switcher-btn {
  width: 120px;
  height: 30px;
  right: 0;
  position: absolute;
  margin-right: 30px;
  background-color: #2f94e0;
  border-radius: 5px;
  border: 0;
  cursor: pointer;
  color: #fff;

  &:focus,
  &:active {
    background-color: #6ca9d7;
  }

  &:hover {
    background-color: #0084e8;
  }
}
