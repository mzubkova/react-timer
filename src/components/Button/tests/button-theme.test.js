import React from "react";
import { shallow, mount } from "enzyme";
import ButtonTheme from "../ButtonTheme";

describe("ButtonTheme", () => {
  it("Should match shapshot", () => {
    const component = shallow(<ButtonTheme />);
    expect(component.html()).toMatchSnapshot();
  });
  it("Should have Button", () => {
    const component = mount(<ButtonTheme />);
    expect(component.find("ButtonTheme")).toHaveLength(1);
  });
  it("Should have button", () => {
    const component = mount(<ButtonTheme />);
    expect(component.find("button")).toHaveLength(1);
  });
});
