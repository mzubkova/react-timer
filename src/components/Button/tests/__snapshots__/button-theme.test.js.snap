// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonTheme Should match shapshot 1`] = `"<button class=\\"switcher-btn\\">Dark theme</button>"`;
